### PR TITLE
add CustomView with RecyclerView

### PR DESCRIPTION
**Pull request by ByJacob**

### CODE REVIEW

File: app/src/main/java/com/fows/custom/RecyclerViewProgressBar.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    @BindView(R.id.progressBar2)
+    ProgressBar progressBar2;
+
+    public RecyclerViewProgressBar(Context context) {
+        this(context, null);
+    }
+
+    public RecyclerViewProgressBar(Context context, @Nullable AttributeSet attrs) {
+        super(context, attrs);
+        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
```
Comment (andrzejchm) [inline app/src/main/java/com/fows/custom/RecyclerViewProgressBar.java]: I'd highly discourage inflating items into `RecyclerView` directly, as this view is not supposed to have any child views other than those provided by adapter. There is a big chance this won't work now or in a future update of `RecyclerView`

File: app/src/main/java/com/fows/custom/RecyclerViewProgressBar.java [hunk 2]
```diff
@@ -15,9 +16,12 @@
  * Created by ByJacob on 2016-11-22-16:46.
  */
 
-public class RecyclerViewProgressBar extends RecyclerView {
-    @BindView(R.id.progressBar2)
-    ProgressBar progressBar2;
+public class RecyclerViewProgressBar extends RelativeLayout {
```
Comment (MBratkowski) [inline app/src/main/java/com/fows/custom/RecyclerViewProgressBar.java]: You should use a `FrameLayout`. It is faster than `RelativeLayout`.

File: app/src/main/java/com/fows/custom/RecyclerViewProgressBar.java [hunk 3]
```diff
@@ -30,10 +34,10 @@ public RecyclerViewProgressBar(Context context, @Nullable AttributeSet attrs) {
     }
 
     public void showLoading() {
-        progressBar2.setVisibility(VISIBLE);
+        progressBar.setVisibility(VISIBLE);
     }
 
     public void hideLoading() {
-        progressBar2.setVisibility(INVISIBLE);
+        progressBar.setVisibility(INVISIBLE);
```
Comment (MBratkowski) [inline app/src/main/java/com/fows/custom/RecyclerViewProgressBar.java]: You should use a `GONE` attribute. When you are use `INVISIBLE` attribute view will hide but it still takes up space for layout purposes. While `GONE`  is invisible, and it doesn't take any space for layout purposes.

**Comment by andrzejchm** [comment]
As far as I can see both solutions are pretty simmilar to the `ViewAnimator`. All you would need to do is: 
```
<ViewAnimator ...>

    <RecyclerView .../>

   <ProgressBar ../>

</ViewAnimator>
```
and switch between childs like

```
viewAnimator.setDisplayedChild(POSITION_RECYCLER_VIEW);

or

viewAnimator.setDisplayedChild(POSITION_PROGRESS_BAR);
```
where `POSITION_RECYCLER_VIEW = 0` and `POSITION_PROGRESS_BAR = 1`

**Comment by andrzejchm** [comment]
`ViewAnimator` solution gives you an advantage of free support for transition animations between views with `android:inAnimation` and `android:outAnimation`

File: app/src/main/java/com/fows/custom/RecyclerViewProgressBar.java (hunk 3 above)
Comment (andrzejchm) [inline app/src/main/java/com/fows/custom/RecyclerViewProgressBar.java]: taking space is not the problem, its measuring the view boundaries which causes a problem with setting view's visibility to `INVISIBLE`.

**Pull request closed without merging**